lib/TikZImage.pm

```diff
@@ -48,7 +48,7 @@ sub new {
 			if ($field eq 'ext') {
 				my $ext = shift;
 				$data->{ext} = $ext
-				if ($ext && ($ext eq 'png' || $ext eq 'gif' || $ext eq 'svg' || $ext eq 'pdf'));
+				if ($ext && ($ext =~ '^(png|gif|svg|pdf|tgz)$'));
```

Comment:
```suggestion
				if ($ext && ($ext =~ /^(png|gif|svg|pdf|tgz)$/));
```
The right hand side of the match operator needs to be a regular expression.

Reply:
Elsewhere you mention regular expressions are slow. Should I switch this back to a bunch of `eq` and `||`?

Reply:
No, this is probably better. Regular expressions are allowed than direct string comparisons, and so if there is a direct string comparison that should be preferred. In this case however, we are comparing a single regular expressions to several string comparisons. This is at least more concise coding wise.

Reply:
I did a quick timing test on this.  So running
```
my $ext = 'png';
for (0 .. 10000000) {
	my $test = $ext eq 'png' || $ext eq 'gif' || $ext eq 'svg' || $ext eq 'pdf' || $ext eq 'tgz';
}
```
took about 1.2 seconds, while runing
```
my $ext = 'png';
for (0 .. 10000000) {
	my $test = $ext =~ /^(png|gif|svg|pdf|tgz)$/;
}
```
took about 4.1 seconds.

Reply:
This code is not going to be executed ten million times though.  It will probably occur once or twice per problem.  So these timing considerations aren't that significant.

Reply:
That's interesting. You made `$ext` be "png" and I wonder if it would affect the timing since "png" is first. In the string comparison, does perl even bother with the subsequent comparisons if the first one works out to true?

Reply:
Good point.  So if I set `my $ext = 'tgz'` then the string comparison executed ten million times takes about 2.5 seconds, and the regular expression takes about 4.1 seconds.  If I set it to `svg` (the middle value) then the string comparison takes about 1.9 seconds ant the regular expression about 3.9 seconds.  So a value later in the list is slower, but the never quite as slow as the regular expression.

Reply:
Here is a more comprehensive test (full script):
```perl
#!/usr/bin/env perl

use strict;
use warnings;

use Time::HiRes qw/time/;

my $start = time;
for (0 .. 1000000) {
	for my $ext ('png', 'gif', 'svg', 'pdf', 'tgz') {
		my $test = $ext eq 'png' || $ext eq 'gif' || $ext eq 'svg' || $ext eq 'pdf' || $ext eq 'tgz';
	}
}
print "string compare: " . (time - $start) . "\n";

$start = time;
for (0 .. 1000000) {
	for my $ext ('png', 'gif', 'svg', 'pdf', 'tgz') {
		my $test = $ext =~ /^(png|gif|svg|pdf|tgz)$/;
	}
}
print "regex compare: " . (time - $start) . "\n";
```
This outputs something like
```
string compare: 1.15723896026611
regex compare: 2.274906873703
```
Several executions gave similar results.